app/policies/repository_policy.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+# Policies for RepositoriesController
+class RepositoryPolicy < ApplicationPolicy
+  attr_reader :current_user, :repository
+
+  def initialize(current_user, repository)
+    @current_user = current_user
+    @repository = repository
+    super
+  end
+
+  def show?
+    repository.public_access == true ||
+      !!current_user&.accessible_repositories&.map(&:to_param)&.include?(repository.to_param)
+  end
+
+  def create?(owner)
+    if owner.is_a?(User)
+      owner&.id == current_user&.id if current_user
+    elsif owner.is_a?(Organization)
+      !OrganizationMembership.find(member: current_user, organization: owner,                                role: 'admin').nil?
```

Comment:
It would be easier to understand this code if you wrote

```ruby
!!OrganizationMembership.find(member: current_user,
                              organization: owner,
                              role: 'admin')
```